### ISSUE TITLE
Extend this to Autodesk Shape Manager

### ISSUE THREAD

**Issue by goatchurchprime** [orbingol/ACIS-Python3#8]
I know this should put this in an email, but then I thought it would be more useful in the long term to post any comments publicly.  

The ASM is a fork of the ACIS kernel (in case you didn't know) and would have more or less the same interface.  (It also uses Scheme.)

When I worked at Autodesk I persuaded someone to finish their SWIG compilation against the ASM, so I had a Python interface almost immediately.   
    https://www.freesteel.co.uk/wpblog/2015/08/03/on-the-shape-of-autodesk/

This would be a more efficient way for ACIS to obtain this feature, if there was an application for it.  

You can tell there are a lot of problems in the CAD industry, when you realize that Dassault has been singularly unable to swap Parasolid for ACIS in Solidworks, when they are so exposed by their rival Siemans owning Parasolid.  BTW all three solid modelling kernels (Parasolid, ACIS and ASM) have their developer teams within a mile of one another in Cambridge, England.  This is a perfect illustration of the unmitigated waste that is caused by closed source development.   We could have had a kernel that was three times better and three times cheaper if it was open source, and it would solve many of the desperate CAD translation problems that waste so much of everyone's time.

**Comment by orbingol** [comment]
Thanks @goatchurchprime 

I know some of the story of Autodesk forking ACIS (I guess ACIS R13?) thanks to the forum that Spatial closed permanently last year. The forum was a valuable resource for me, even though nobody was answering the questions lately.

I also have a semi-working SWIG wrapper for ACIS R26 but I had to do a lot of tricks to make it partially work (and I feel that the reason is that I only have the runtime and the distributed headers, not the fork itself) and it only works with a specific version of SWIG. I could tell tons of stories about ACIS, its bugs and issues that I encountered, SWIG and Boost.Python but I don't think it would be good to expose these publicly. I've never had the chance to try pybind11 and some people I met in the conferences say that it is way better than SWIG and Boost.Python. Doing everything with Python's native C API was a personal experiment for me and I feel that I somehow was able to create an extensible working Python wrapper for ACIS.

It won't be correct, if I say everything is perfect here. The code in this repository still needs a lot of extra work. I know it is already messy (same code everywhere, not the perfect application of Python C API, requires C++ to C mapping and so) and it definitely requires some documentation, especially on how to create a license file, but it somehow works on Windows and Ubuntu LTS variants.

Enough from my side. I completely agree with you about the issues of the industry. I wish there could be a way to fix it, but developing another kernel wouldn't do that. Somehow, these teams need to sit together and work it out but I feel that it is also impossible due to corporate politics.  I think Spatial has some recent work on running ACIS or CGM on the web but for some reason, they killed its web page.

Since the discussion is about open-source, I think you have already heard about openCASCADE. The good thing with openCASCASE is that it comes with a SWIG interface (http://www.pythonocc.org/) instead of Scheme or C#.
